### DISCUSSION TITLE
v20.1.0-rc5

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3321] [Releases]
## You should absolutely know
Since the approval of [our second RFC - release schedule](https://github.com/OpenMage/rfcs/blob/main/accepted/0002-release-schedule.md)  OpenMage 19.x enters and "patch only" state, it will be maintained for two more years as promised but only _significant security patches or regression fixes_ will be ported to v19, every other development (and we have many) will be focused on v20+.

We encourage everybody to upgrade to v20, it is our latest and greatest and deserves the bit of work necessary for the upgrade (ask your developer/agency, don't do it yourself).

## Release highlights

This is a big release, that's why we decided to move away from the 20.0.x versioning and go to 20.1.x. Since a lot of changes could have some impact on current installations we decided to release some "rc" versions before the official 20.1.0. Tests are more than welcome now but be extra careful with production environment.

What's most important is the removal of all the 3rd party libraries (phpseclib, mcrypt_compat, Cm_RedisSession, Cm_Cache_Backend_Redis and Pelago_Emogrifier and Zend Framework) form our repository, they are now imported via composer. This was an important step to clean up and modernise our code.
* If you want to clean up your project from all the files that were removed during this release cycle, please [check this link](https://github.com/OpenMage/magento-lts/discussions/3170#discussioncomment-5722151) for the complete list.

Don't worry though, if you've always installed OpenMage extracting the zip file, starting from this release you'll find a new zip file attached to the release itself, we build this zip adding all of the old 3rd party libraries so that you will not have to migrate to composer or use composer at all.

Also:
- PHP 7.4 is now the minimum required version and 8.2 is now supported.
- the M1 legacy themes have been moved to an external repository since it's old (and mostly unused) code.
- a great improvement to EAV config cache has been added to v20.
- support for Google Analytics 4 was added with enhanced ecommerce data (which historically was always provided by 3rd party extensions)
- possibility to set backend locale per every admin user was added.
- "pub/" folder structure was introduced (optional), to have extra security for your installation
- massive performance improvement in complex SQL queries (#2210)
- many, many smaller bugfix and improvements

## Changelog
* Minor phpstan-fixes for php8.1 by @sreichel in https://github.com/OpenMage/magento-lts/pull/3237
* Fixed the behaviour of $forceLog parameter for Mage::log() by @empiricompany in https://github.com/OpenMage/magento-lts/pull/3263
* Added support for Google Analytics 4 Enhanced eCommerce by @discountscott in https://github.com/OpenMage/magento-lts/pull/3198
* Optimized count queries by removing all joins if there are no where by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/2210
* Added PHP8 configurations to .htaccess by @fballiano in https://github.com/OpenMage/magento-lts/pull/3254
* GitHub Actions - `check-files` workflow - split into steps as a workaround for step template size limit by @gorbunovav in https://github.com/OpenMage/magento-lts/pull/3276
* Fix admin ACL mismatch for OAuth (#3272) by @Tomasz-Silpion in https://github.com/OpenMage/magento-lts/pull/3274
* Fixed PHP8.2 deprecated warning in Mage_Cron_Model_Observer by @fballiano in https://github.com/OpenMage/magento-lts/pull/3253
* Introduced pub/ structure for webserver home folder mapping by @speedupmate in https://github.com/OpenMage/magento-lts/pull/1210
* Bump tj-actions/changed-files from 35 to 36 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3286
* Bump friendsofphp/php-cs-fixer from 3.16.0 to 3.17.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3285
* Fixed deprecation warning in Mage_Core_Model_Url::isOwnOriginUrl() by @fballiano in https://github.com/OpenMage/magento-lts/pull/3284
* Escape product names and translations inside js by @Judx in https://github.com/OpenMage/magento-lts/pull/2673
* Removed auto_detect_line_endings since it's deprecated in 8.1 by @fballiano in https://github.com/OpenMage/magento-lts/pull/3282
* Replaced strpos() calls with PHP8 string functions by @sreichel in https://github.com/OpenMage/magento-lts/pull/3258
* Fix regression from getSelectCountSql optimization when collection us… by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3279
* Mage_Rules now truncates strings after 100 characters instead of 33 in the backend by @loekvangool in https://github.com/OpenMage/magento-lts/pull/3288
* Added back Order add comment functionality to REST API by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3292
* Fixed E_DEPRECATED str_replace(): Passing null in Translate.php by @kiatng in https://github.com/OpenMage/magento-lts/pull/3294
* Fixed passing null is deprecated for trim in Mage_Tax_Model_Resource_Calculation by @kyrena in https://github.com/OpenMage/magento-lts/pull/3299
* Fixed documentation on how to contribute after branch change by @fballiano in https://github.com/OpenMage/magento-lts/pull/3306
* Fixed passing null is deprecated for stripos in Mage_Core_Model_Resource_Helper_Abstract by @kyrena in https://github.com/OpenMage/magento-lts/pull/3295
* Fixed passing null is deprecated for explode in Mage_Catalog_Model_Resource_Product_Indexer_Eav_Source by @kyrena in https://github.com/OpenMage/magento-lts/pull/3297
* Allowed system config of type "serialized array" to <depend> on another field by @pquerner in https://github.com/OpenMage/magento-lts/pull/3307
* Fixed passing null is deprecated for htmlspecialchars in Mage_Adminhtml_Block_System_Config_Form_Field by @kyrena in https://github.com/OpenMage/magento-lts/pull/3296
* "Product Tags" tab is hidden in customer admin mask if Mage_Tag is disabled by @kyrena in https://github.com/OpenMage/magento-lts/pull/3298
* Fix #3312 ExtJS Tree Checkbox Update by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/3314
* Bump phpunit/phpunit from 9.6.8 to 9.6.9 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3317
* Bump phpstan/phpstan from 1.10.15 to 1.10.18 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3316
* Fixed display of products count for categories tree by @kyrena in https://github.com/OpenMage/magento-lts/pull/3313
* Some PHPStan fixes for the lib/Varien folder by @fballiano in https://github.com/OpenMage/magento-lts/pull/3271
* Added .gitattributes file by @tmotyl in https://github.com/OpenMage/magento-lts/pull/3318
* Enhanced Google Analytics 4 Integration and Ecommerce Tracking Improvements by @empiricompany in https://github.com/OpenMage/magento-lts/pull/3305
* Added few missing export-ignore to gitattributes by @tmotyl in https://github.com/OpenMage/magento-lts/pull/3320

## New Contributors
* @gorbunovav made their first contribution in https://github.com/OpenMage/magento-lts/pull/3276
* @Tomasz-Silpion made their first contribution in https://github.com/OpenMage/magento-lts/pull/3274
* @speedupmate made their first contribution in https://github.com/OpenMage/magento-lts/pull/1210
* @Judx made their first contribution in https://github.com/OpenMage/magento-lts/pull/2673
* @kyrena made their first contribution in https://github.com/OpenMage/magento-lts/pull/3299
* @pquerner made their first contribution in https://github.com/OpenMage/magento-lts/pull/3307

**Full Changelog**: https://github.com/OpenMage/magento-lts/compare/v20.1.0-rc4...v20.1.0-rc5

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v20.1.0-rc5'>v20.1.0-rc5</a>.</em>

**Comment by S0FTWEX** [comment]
Because of #3258, the minimal required version of PHP is 8.0.

**Reply by sreichel** [reply]
Please see #2946
